### ISSUE TITLE
Rancher CLI - Error on installing an app when the default value for the catalog app is not applicable for rancher version

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#22127]
**What kind of request is this (question/bug/enhancement/feature request):** bug

**Steps to reproduce (least amount of steps as possible):**
Setup details:
```
rancher version: v2.3.2(custom image for testing)
chart - fluentd-aggregator
chart version: latest - rancher_min_version: 2.3.2 and rancher_max_version: 2.4.0
chart version: v0.1.0 - rancher_max_version: 2.3.2
chart version: v0.2.1 - rancher_min_version: 2.3.3
```
Download rancher CLI
Login to rancher v2.3.2 and switch context to your project.
./rancher app install c-npslj:sowmya-cluster-fluentd-aggregator

**Expected Result:**
app with `v0.1.0` should be deployed, and installed successfully.

**Actual Result:**
Error `FATA[0005] Bad response statusCode [422]. Status [422 Unprocessable Entity]. Body: [message=rancher min version not met, baseType=error, code=InvalidBodyContent] from [https://x.x.x.x/v3/project/c-npslj:p-dkzlv/apps] ` is seen

This is because the default app version considered is `0.3.0`

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): rancher - master
- Installation option (single install/HA): single

**Comment by dramich** [comment]
Available in v2.3.0-rc4

**Comment by sowmyav27** [comment]
verified with `v2.3.0-rc4`, and does not work as expected.
```
$./rancher app st c-npslj:sowmya-cluster-fluentd-aggregator
0.1.0
0.1.2
0.1.6
$ ./rancher app install c-npslj:sowmya-cluster-fluentd-aggregator
FATA[0000] Bad response statusCode [404]. Status [404 Not Found]. Body: [baseType=error, code=NotFound, message=failed to find resource by id] from [https://x.x.x.x/v3/templateversions/0.1.6] 
```
The app fails to install with the default template version.
But when the version is mentioned explicitly the app installs successfully.
```
./rancher app install c-npslj:sowmya-cluster-fluentd-aggregator --version 0.1.6
```

**Comment by dramich** [comment]
Available in v2.3.0-rc5

**Comment by sowmyav27** [comment]
verified with `v2.3.0-rc5`, and works as expected.
```
$./rancher app st <app-name>
0.1.0
0.1.2
0.1.6

$ ./rancher app install <app-name>
installs the default app version - 0.1.6

$ ./rancher app install --version 0.1.6 <app-name>
```
